### ISSUE TITLE
Persist best epoch per trial

### ISSUE THREAD

**Issue by rakshithvsk** [keras-team/tf-keras#367]
**System information**.

TensorFlow version (you are using): **2.9.2**

**Feature and the current behavior/state**
Currently, For train_end() method of keras callback, the output of the last call to `on_epoch_end()` is saved as logs, though it is not the best epoch for a given trial. Apparently it must be the output of best of all epochs. 

**Description**
1. Why it would be helpful to have the best epoch to be passed to train_end() method is because, at the end of the trial all we would be interested in, is the best metrics only with given objectives to monitor(val_f1/f1/loss..). 
2. Currently its misleading to see the latest epoch metrics being logged, **even though its not the best one**
3. This is causing major issues while integrating with platforms like MLFLow using callbacks, because for every trial the latest epoch's model and metrics are stored, **though they are not the best ones** among all the epochs for a particular trial.
4. On top of it, it would be seamless to log and view the best metrics for a trial. I don't see a way currently to view the best epoch and metrics for a given trial.

**Will this change the current api? How?**
No

**Who will benefit from this feature?**
Everyone who is keen on logging and monitoring the metrics for every trial

**Comment by tilakrayal** [comment]
Have you tried to create a custom callback by overriding on_epoch_end to track your desired "best" metric. Then, in on_train_end, access this stored state. ModelCheckpoint offers a good example, but your implementation can be much simpler.

Thank you!

**Comment by sachinprasadhs** [comment]
Thank you for reporting the issue. 
We're now prioritizing [Keras 3](https://keras.io/keras_3/) development, a multi-backend framework that offers significant advantages by supporting `JAX`, `TensorFlow`, `PyTorch` and `OpenVINO`. 
This means we won't be adding new features to `TF-Keras (Keras 2)`. 

`TensorFlow` users can benefit directly from `Keras 3`, which is included by default starting with version `2.16.1`.
 We encourage you to migrate to `Keras 3` and experience its enhanced capabilities. If you feel this specific feature is crucial for the `Keras` framework itself, please open a new issue at: https://github.com/keras-team/keras/issues

For more details and usage guide about Keras 3, refer: https://keras.io/getting_started/about/
